omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierResource.java

```diff
@@ -61,11 +87,56 @@ public void purge(String parentUniqueId, String uniqueId, RequestContext context
 
     @Override
     public SimpleObject getAll(String parentUniqueId, RequestContext context) throws ResponseException {
-        throw new UnsupportedOperationException();
+        throw new ResourceDoesNotSupportOperationException();
     }
 
     @Override
     public void put(String parentUniqueId, SimpleObject post, RequestContext context) throws ResponseException {
-        throw new UnsupportedOperationException();
+        throw new ResourceDoesNotSupportOperationException();
     }
+
+	@Override
+	public Identifier newDelegate() {
+		throw new ResourceDoesNotSupportOperationException();
+	}
+
+	@Override
+	public Identifier save(Identifier delegate) {
+		throw new ResourceDoesNotSupportOperationException();
+	}
+
+	@Override
+	public IdentifierSource getParent(Identifier instance) {
+		throw new ResourceDoesNotSupportOperationException();
+	}
+
+	@Override
+	public void setParent(Identifier instance, IdentifierSource parent) {
+		throw new ResourceDoesNotSupportOperationException();
+	}
+
+	@Override
+	public PageableResult doGetAll(IdentifierSource parent, RequestContext context) throws ResponseException {
+		throw new ResourceDoesNotSupportOperationException();
+	}
+
+	@Override
+	public Identifier getByUniqueId(String uniqueId) {
+		throw new ResourceDoesNotSupportOperationException();
+	}
+
+	@Override
+	protected void delete(Identifier delegate, String reason, RequestContext context) throws ResponseException {
+		throw new ResourceDoesNotSupportOperationException();
+	}
+
+	@Override
+	public void purge(Identifier delegate, RequestContext context) throws ResponseException {
+		throw new ResourceDoesNotSupportOperationException();		
+	}
+
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		throw new UnsupportedOperationException();
```

Comment:
Are you sure we do not support all these operations?

Reply:
Yes it's trivial overriding this since we don't support any get operations. We only support POST for generating Identifiers. 
BTW, whats the significancy of overriding this method, just to insight me!

Reply:
Any reason for mixing ResourceDoesNotSupportOperationException and UnsupportedOperationException?